File: src/main/java/net/atos/zac/app/productaanvragen/converter/RESTInboxProductaanvraagListParametersConverter.java
/*
 * SPDX-FileCopyrightText: 2022 Atos
 * SPDX-License-Identifier: EUPL-1.2+
 */

package net.atos.zac.app.productaanvragen.converter;

import net.atos.zac.app.productaanvragen.model.RESTInboxProductaanvraagListParameters;
import net.atos.zac.app.shared.RESTListParametersConverter;
import net.atos.zac.productaanvraag.model.InboxProductaanvraagListParameters;
import nl.info.zac.search.model.DatumRange;

public class RESTInboxProductaanvraagListParametersConverter extends
                                                             RESTListParametersConverter<InboxProductaanvraagListParameters, RESTInboxProductaanvraagListParameters> {

    @Override
    protected void doConvert(
            final InboxProductaanvraagListParameters listParameters,
            final RESTInboxProductaanvraagListParameters restListParameters
    ) {
        listParameters.setType(restListParameters.type);
        listParameters.setInitiatorID(restListParameters.initiatorID);

        if (restListParameters.ontvangstdatum != null && restListParameters.ontvangstdatum.hasValue()) {
            listParameters.setOntvangstdatum(new DatumRange(restListParameters.ontvangstdatum.getVan(), restListParameters.ontvangstdatum
                    .getTot()));
        }
    }

    @Override
    protected InboxProductaanvraagListParameters getListParameters() {
        return new InboxProductaanvraagListParameters();
    }
}

File: src/main/kotlin/nl/info/zac/app/productaanvragen/converter/RestInboxProductaanvraagListParametersConverter.kt
/*
 * SPDX-FileCopyrightText: 2022 Atos, 2026 INFO.nl
 * SPDX-License-Identifier: EUPL-1.2+
 */

package nl.info.zac.app.productaanvragen.converter

import nl.info.zac.app.productaanvraag.model.RestInboxProductaanvraagListParameters
import nl.info.zac.app.shared.applyCommonParametersTo
import nl.info.zac.productaanvraag.model.InboxProductaanvraagListParameters
import nl.info.zac.search.model.DatumRange

fun RestInboxProductaanvraagListParameters.toInboxProductaanvraagListParameters() =
    InboxProductaanvraagListParameters().also { params ->
        this.applyCommonParametersTo(params)
        params.type = this.type
        params.initiatorID = this.initiatorID
        this.ontvangstdatum?.let { ontvangstdatum ->
            if (ontvangstdatum.hasValue()) {
                params.ontvangstdatum = DatumRange(van = ontvangstdatum.van, tot = ontvangstdatum.tot)
            }
        }
    }
